src/mono/mono/mini/interp/transform.c

```diff
@@ -1479,6 +1479,37 @@ interp_handle_intrinsics (TransformData *td, MonoMethod *target_method, MonoClas
 	} else if (in_corlib && !strcmp (klass_name_space, "System") && !strcmp (klass_name, "ByReference`1")) {
 		g_assert (!strcmp (tm, "get_Value"));
 		*op = MINT_INTRINS_BYREFERENCE_GET_VALUE;
+	} else if (in_corlib && !strcmp (klass_name_space, "System") && !strcmp (klass_name, "Marvin")) {
+		if (!strcmp (tm, "Block"))
+			*op = MINT_INTRINS_MARVIN_BLOCK;
+	} else if (in_corlib && !strcmp (klass_name_space, "System.Text.Unicode") && !strcmp (klass_name, "Utf16Utility")) {
```

Comment:
Is this going to silently regress perf if Utf16Utility is ever refactored and things in it shuffled/renamed? This is not a public API so the risk of that is pretty high.

Cc @GrabYourPitchforks

Reply:
Unfortunately yes. We should spot potential regressions with the daily perf runs that we will do. Hopefully we will not make use of this technique on a large enough scale for it to become problematic.

Reply:
can we mark the method with `[Intrinsic]` or add some comment to make the chance of this happening a bit less likely?